src/core/automount.c

```diff
@@ -592,7 +611,17 @@ static void automount_enter_waiting(Automount *a) {
         if (r < 0)
                 goto fail;
 
-        xsprintf(options, "fd=%i,pgrp="PID_FMT",minproto=5,maxproto=5,direct", p[1], getpgrp());
+
+        if (asprintf(
+                    &options,
+                    "fd=%i,pgrp="PID_FMT",minproto=5,maxproto=5,direct,%s",
+                    p[1],
+                    getpgrp(),
+                    user_options) < 0) {
+                r = -ENOMEM;
+                goto fail;
+        }
```

Comment:
this will add a trailing "," to the end of the string if user_options is NULL

Reply:
AutoFS ignores empty k/v pairs like this, so it didn't feel worth it to complicate the code here.

Reply:
We still want this to be clean,

use strextendf_with_separator() or so, to nicel avoid the extra ","